packages/shell-api/src/mongo.ts

```diff
@@ -174,4 +171,24 @@ export default class Mongo extends ShellApiClass {
   async setReadConcern(level: string): Promise<void> {
     await this._serviceProvider.resetConnectionOptions({ readConcern: { level: level } });
   }
+
+  startSession(options: SessionOptions = {}): Session {
+    return new Session(this, options || {} as SessionOptions, this._serviceProvider.startSession(options));
+  }
+
+  setCausalConsistency(): void {
+    throw new MongoshInvalidInputError('It is not possible to set causal consistency for an entire connection due to the driver, use startSession({causalConsistency: <>}) instead.');
+  }
+
+  isCausalConsistency(): void {
+    throw new MongoshInvalidInputError('Causal consistency for drivers is set via Mongo.startSession and can be checked via session.getOptions. The default value is true');
+  }
+
+  setSlaveOk(): void {
+    throw new MongoshInvalidInputError('setSlaveOk is deprecated.');
```

Comment:
Is this also being replaced by `setReadPreference`?

```suggestion
    throw new MongoshInvalidInputError('setSlaveOk is deprecated, use setReadPreference instead.');
```

Reply:
setting slaveOk and setting read preference are slightly different operations, and from what I could tell / what Matt indicated, there's no straightforward way to set slaveOk using readPreference that might not end up with some unexpected behavior (for example just setting readPreference to secondaryOk isn't the same as setting primaryPreferred)